controllers/cluster_controller_test.go

```diff
@@ -404,201 +275,169 @@ func TestUpgradeOSVersion(t *testing.T) {
 			Namespace: "default",
-			Name:      "np02",
+			Name:      name,
 			Labels: map[string]string{
 				CAPIWatchFilterLabel:  "v0.3.10",
-				capi.ClusterLabelName: "my-cluster",
+				capi.ClusterLabelName: cluster,
 			},
 		},
 		Spec: capiexp.MachinePoolSpec{
-			ClusterName: cluster.Name,
+			ClusterName: cluster,
 			Template: capi.MachineTemplateSpec{
-				ObjectMeta: capi.ObjectMeta{},
 				Spec: capi.MachineSpec{
-					ClusterName: cluster.Name,
-					InfrastructureRef: v12.ObjectReference{
-						Kind:       azureMachinePool2.Kind,
-						Name:       azureMachinePool2.Name,
-						APIVersion: azureMachinePool2.APIVersion,
-					},
-					Version: to.StringPtr("v1.18.2"),
+					ClusterName: cluster,
+					Version:     to.StringPtr("v1.18.2"),
 				},
 			},
 		},
 	}
+}
 
-	ctrlClient := fake.NewFakeClientWithScheme(scheme, release10dot0, azureMachineTemplate, kubeadmcontrolplane, azureCluster, cluster, azureMachinePool1, machinePool1, azureMachinePool2, machinePool2)
-
-	reconciler := ClusterReconciler{
-		Client: ctrlClient,
-		Log:    ctrl.Log.WithName("controllers").WithName("Cluster"),
-		Scheme: scheme,
-	}
-
-	_, err = reconciler.Reconcile(reconcile.Request{NamespacedName: types.NamespacedName{Namespace: "default", Name: "my-cluster"}})
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	// Assert KubeadmControlPlane version is correct.
-	reconciledControlplane := &kcp.KubeadmControlPlane{}
-	err = ctrlClient.Get(ctx, client.ObjectKey{Namespace: kubeadmcontrolplane.Namespace, Name: kubeadmcontrolplane.Name}, reconciledControlplane)
-	if err != nil {
-		t.Fatal(err)
-	}
-	assert.Equal(t, "v1.18.2", reconciledControlplane.Spec.Version, "Kubeadmcontrolplane.Spec.Version uses wrong k8s version")
-
-	foundProviderFile := false
-	expectedProviderFile := fmt.Sprintf("%s-azure-json", reconciledControlplane.Spec.InfrastructureTemplate.Name)
-	for _, file := range reconciledControlplane.Spec.KubeadmConfigSpec.Files {
-		if file.ContentFrom.Secret.Name == expectedProviderFile {
-			foundProviderFile = true
-		}
-	}
-	if !foundProviderFile {
-		t.Fatalf("None of the defined files match the infrastructure machine template name. The name of the provider file needs to match the infrastructure machine template name, got these files %v, expected name %q", reconciledControlplane.Spec.KubeadmConfigSpec.Files, expectedProviderFile)
-	}
+func newAzureMachinePoolChain(cluster string) (*capiexp.MachinePool, *capzexp.AzureMachinePool) {
+	mp := newMachinePool(cluster)
+	amp := newAzureMachinePool(cluster, mp.Name)
 
-	// Assert AzureMachineTemplate used by KubeadmControlPlane uses right machine image.
-	newAzureMachineTemplate := &capz.AzureMachineTemplate{}
-	err = ctrlClient.Get(ctx, client.ObjectKey{Namespace: reconciledControlplane.Spec.InfrastructureTemplate.Namespace, Name: reconciledControlplane.Spec.InfrastructureTemplate.Name}, newAzureMachineTemplate)
-	if err != nil {
-		t.Fatal(err)
+	mp.Spec.Template.Spec.InfrastructureRef = v12.ObjectReference{
```

Comment:
oh this `v12` slip through the crakcs. Let's use an import alias instead of `v12`

Reply:
Yeah, I noticed it and was going to do it but I thought the diff may get too large, but if that's OK I'll go ahead and do it 😀

Reply:
Done.